js/libs/ui-shared/src/utils/darkMode.ts

```diff
@@ -0,0 +1,19 @@
+const DARK_MODE_CLASS = "pf-v5-theme-dark";
+const mediaQuery = window.matchMedia("(prefers-color-scheme: dark)");
+
+function updateDarkMode(isEnabled: boolean = false) {
+  const { classList } = document.documentElement;
+
+  if (isEnabled) {
+    classList.add(DARK_MODE_CLASS);
+  } else {
+    classList.remove(DARK_MODE_CLASS);
+  }
+}
+
+export function initializeDarkMode() {
+  updateDarkMode(mediaQuery.matches);
+  mediaQuery.addEventListener("change", (event: MediaQueryListEvent) =>
```

Comment:
I believe you can omit the type here as it can be inferred.
```suggestion
  mediaQuery.addEventListener("change", (event) =>
```